File: server_asp/src/models/OrderItem.cs
using System;
using System.Collections.Generic;

namespace shieldtify.models;

public partial class OrderItem
{
    public Guid Itemid { get; set; }

    public Guid Orderid { get; set; }

    public int Quantity { get; set; }

    public int SalesPrice { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual Item Item { get; set; } = null!;

    public virtual Order Order { get; set; } = null!;
}

File: server_nodejs/src/models/order_item.js
// Order_item (itemID, orderID, quantity, sales_price)

module.exports = (sequelize, Sequelize) => {
    class OrderItem extends Sequelize.Model {}

    OrderItem.init(
        {
            itemid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            orderid: {
                type: Sequelize.UUID,
                primaryKey: true,
            },
            quantity: {
                type: Sequelize.INTEGER,
                allowNull: false,
            },
            sales_price: {
                type: Sequelize.FLOAT,
                allowNull: false,
            },
        },
        {
            sequelize,
            modelName: 'order_item',
            timestamps: true,
            paranoid: false,
            underscored: true,
        },
    );

    OrderItem.associate = (models) => {
        OrderItem.belongsTo(models.item, {
            foreignKey: 'itemid',
        });
        OrderItem.belongsTo(models.order, {
            foreignKey: 'orderid',
        });
    };

    return OrderItem;
};
